### PR TITLE
API changes

### PR DESCRIPTION
**Pull request by adasq**
Another set of changes for the public API engine.

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -1,129 +1,197 @@
...
+    };
+    actionHandlers[action.type] && actionHandlers[action.type]();
+}
+
+function onPhaseUpdated(next: Function, phaseInit: boolean) {
+    const state: Game = thousand.getState();
+
+    const phaseHandlers = {
+        [Phase.REGISTERING_PLAYERS_START]: () => {
+            console.log('show registering player view');
```
Comment (alan-null) [inline main.ts]: Maybe we could use some abstraction for logging? So you easily change the log level or disable it completely.

File: main.ts [hunk 2]
```diff
@@ -1,129 +1,197 @@
...
-        wonCards: {
-            'adam': [ ...createCards(8) ],
-            'alan': [ ...createCards(8) ],
-            'pic': [ ...createCards(8) ]
-        }
-    } as Battle
-};
-
-const thousand: Thousand = initializeGame(defaultState);
+// const battleFinishedState: Game = {
```
Comment (alan-null) [inline main.ts:9]: I am against commented code. If it is commended it is not needed?

File: main.ts (hunk 1 above)
Comment (alan-null) [inline main.ts:112]: Maybe we could use some abstraction for logging? So you easily change the log level or disable it completely.

Comment (adasq) [reply main.ts:112]: main.js is an API showcase.

File: main.ts (hunk 2 above)
Comment (adasq) [reply main.ts:9]: I will move it later to another file, main.js is an API showcase, so it's more how to use 1k rather than logic itself

**Pull request merged**